src/WebJobs.Script.WebHost/Diagnostics/Extensions/ScriptHostServiceLoggerExtension.cs

```diff
@@ -0,0 +1,123 @@
+        {
+            logger.Log(LogLevel.Information, _initializing, "Initializing Host.");
+        }
+
+        public static void Initialization(this ILogger logger, int attemptCount, int startCount)
+        {
+            logger.Log(LogLevel.Information, _initialization, $"Host initialization: ConsecutiveErrors={attemptCount}, StartupCount={startCount}");
+        }
+
+        public static void InStandByMode(this ILogger logger)
+        {
+            logger.Log(LogLevel.Information, _inStandByMode, "Host is in standby mode");
+        }
+
+        public static void UnhealthyRestart(this ILogger logger)
+        {
+            logger.Log(LogLevel.Error, _unhealthyRestart, "Host is unhealthy. Initiating a restart.");
+        }
+
+        public static void Stopping(this ILogger logger)
+        {
+            logger.Log(LogLevel.Information, _stopping, "Stopping host...");
+        }
+
+        public static void DidNotShutDown(this ILogger logger)
+        {
+            logger.Log(LogLevel.Warning, _didNotShutDown, "Host did not shutdown within its allotted time.");
+        }
+
+        public static void ShutDownCompleted(this ILogger logger)
+        {
+            logger.Log(LogLevel.Information, _shutDownCompleted, "Host shutdown completed.");
+        }
+
+        public static void SkipRestart(this ILogger logger, ScriptHostState state)
+        {
+            logger.Log(LogLevel.Debug, _skipRestart, $"Host restart was requested, but current host state is '{state}'. Skipping restart.");
+        }
+
+        public static void Restarting(this ILogger logger)
+        {
+            logger.Log(LogLevel.Information, _restarting, "Restarting host.");
+        }
+
+        public static void Restarted(this ILogger logger)
+        {
+            logger.Log(LogLevel.Information, _restarted, "Host restarted.");
+        }
+
+        public static void Building(this ILogger logger, bool skipHostStartup, bool skipHostJsonConfiguration)
+        {
+            logger.Log(LogLevel.Information, _building, $"Building host: startup suppressed:{skipHostStartup}, configuration suppressed: {skipHostJsonConfiguration}");
+        }
+
+        public static void StartupWasCanceled(this ILogger logger)
+        {
+            logger.Log(LogLevel.Debug, _startupWasCanceled, "Host startup was canceled.");
+        }
+
+        public static void ErrorOccured(this ILogger logger, Exception ex)
+        {
+            logger.Log(LogLevel.Debug, _errorOccured, ex, "A host error has occurred");
+        }
+
+        public static void ErrorOccuredInactive(this ILogger logger, Exception ex)
+        {
+            logger.Log(LogLevel.Debug, _errorOccuredInactive, ex, "A host error has occurred on an inactive host");
+        }
+
+        public static void CancellationRequested(this ILogger logger)
+        {
+            logger.Log(LogLevel.Debug, _cancellationRequested, "Cancellation requested. A new host will not be started.");
+        }
+    }
+}
```

Comment:
We should use LoggerMessage for this -- https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/loggermessage?view=aspnetcore-2.2. I know we already use them somewhere else, but they'd be a great fit here as well.

Reply:
I think you're talking about the pattern used here: https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script.WebHost/Middleware/HostAvailabilityCheckMiddleware.Log.cs